make build-node-image fails with Ansible requires 3.9.0 or greater.

<!--
Thank you for the Bug report! Before submitting, please delete all of
the HTML comments.
-->

**What steps did you take and what happened:**

```
# make build-node-image OS_TARGET=ubuntu-2004-efi KUBERNETES_VERSION=v1.26.5+vmware.2-fips.1 TKR_SUFFIX=gpu ARTIFACTS_CONTAINER_IP=192.168.30.95 IMAGE_ARTIFACTS_PATH=/root/image ARTIFACTS_CONTAINER_PORT=8081
DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
            Install the buildx component to build images with BuildKit:
            https://docs.docker.com/go/buildx/

Sending build context to Docker daemon   31.2MB
Step 1/26 : FROM photon:3.0
 ---> 42ffa214b8d3
<snip>
....

Cloning into 'image-builder'...
Removing intermediate container 1324a8a3ce48
 ---> fe4910d02848
Step 18/26 : WORKDIR $IMAGE_BUILDER_REPO_NAME
 ---> Running in 994e9b92f18b
Removing intermediate container 994e9b92f18b
 ---> c3f9ed5b598f
Step 19/26 : WORKDIR images/capi
 ---> Running in 2b8c0347c5d3
Removing intermediate container 2b8c0347c5d3
 ---> 816a749a884b
Step 20/26 : RUN make deps-ova
 ---> Running in 86e1292f295f
hack/ensure-python.sh
Checking if python is available
Detected python version: Python 3.7.5.
Ansible requires 3.9.0 or greater.
Please install 3.9.0 or later.
make: *** [Makefile:59: deps-common] Error 2
The command '/bin/bash -c make deps-ova' returned a non-zero code: 2
make: *** [Makefile:69: build-image-builder-container] Error 2
```

**What did you expect to happen:**
Build to be successful

**Please tell us about your environment.**

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,10 +1,10 @@
 # Copyright 2023 VMware, Inc.
 # SPDX-License-Identifier: MPL-2.0
 
-FROM photon:3.0
+FROM photon:4.0
 
-ARG PACKER_VERSION=1.9.1
-ARG ANSIBLE_VERSION=2.11.5
+ARG IMAGE_BUILDER_COMMIT_ID=""
+ARG ANSIBLE_VERSION=2.12.8
 ARG IMAGE_BUILDER_REPO_NAME=image-builder
 
 ENV PATH=${PATH}:/ovftool
@@ -15,18 +15,10 @@ RUN tdnf -y update
 RUN tdnf -y upgrade
 
 # Install required packages
-RUN for package in unzip git wget build-essential python3.8 python3-pip python-jinja2 jq coreutils openssh-server; do tdnf -y install "$package"; done
-
-# Install Packer
-RUN wget https://releases.hashicorp.com/packer/"$PACKER_VERSION"/packer_"$PACKER_VERSION"_linux_amd64.zip
-RUN unzip packer_"$PACKER_VERSION"_linux_amd64.zip -d /tmp/
-RUN mv /tmp/packer /usr/local/bin/
-
-# Install Ansible
-RUN pip3 install ansible-core==$ANSIBLE_VERSION
+RUN for package in unzip git wget build-essential python3-pip jq coreutils openssh-server; do tdnf -y install "$package"; done
 
 # Install Semver
-RUN pip3 install semver
+RUN pip3 install semver jinja2
 
 # Install ovftool
 # TODO: this URL might change or expire so need to look at better way to install it on the container.
@@ -37,7 +29,14 @@ RUN unzip VMware-ovftool-4.4.3-18663434-lin.x86_64.zip -d /
 RUN git clone https://github.com/kubernetes-sigs/image-builder.git
 WORKDIR $IMAGE_BUILDER_REPO_NAME
 
-# Running deps-ova to setup packer goss provisioner
+RUN git checkout $IMAGE_BUILDER_COMMIT_ID
+
+# Install Ansible
+RUN pip3 install ansible-core==$ANSIBLE_VERSION
+# Set the environment variable where packer will be installed
+ENV PATH=${PATH}:/image-builder/images/capi/.local/bin
+
+# Running deps-ova to setup packer, goss provisioner and other ansible galaxy collections
 WORKDIR images/capi
 RUN make deps-ova
 

--- a/Makefile
+++ b/Makefile
@@ -7,7 +7,6 @@ MAKEFLAGS += -s
 .EXPORT_ALL_VARIABLES:
 
 # Default variables
-BYOI_IMAGE_NAME = vsphere-tanzu-byoi
 DEFAULT_ARTIFACTS_CONTAINER_PORT = 8081
 DEFAULT_PACKER_HTTP_PORT = 8082
 MAKE_HELPERS_PATH = $(shell pwd)/hack/make-helpers
@@ -57,8 +56,11 @@ define BUILD_IMAGE_BUILDER_CONTAINER_HELP_INFO
 # Will create a docker container image for creation of TKGs OVA with the dependencies
 # like packer, ansible and kubernetes image builder code.
 # 
+# Arguments:
+#   KUBERNETES_VERSION: [Required] kubernetes version of the artifacts containers, to see
+#                       list of supported kubernetes versions use "make list-versions".
 # Example:
-# make build-image-builder-container
+# make build-image-builder-container KUBERNETES_VERSION=v1.23.15+vmware.1
 endef
 .PHONY: build-image-builder-container
 ifeq ($(PRINT_HELP),y)
@@ -66,7 +68,7 @@ build-image-builder-container:
 	printf "$$green$$BUILD_IMAGE_BUILDER_CONTAINER_HELP_INFO$$clear\n"
 else
 build-image-builder-container:
-	docker build --platform=linux/amd64 -q -t $(BYOI_IMAGE_NAME) .
+	$(MAKE_HELPERS_PATH)/build-image-builder-container.sh
 endif
 
 define BUILD_NODE_IMAGE

--- a/README.md
+++ b/README.md
@@ -105,11 +105,11 @@ make run-artifacts-container PRINT_HELP=y                                       
 make run-artifacts-container KUBERNETES_VERSION=v1.22.13+vmware.1 ARTIFACTS_CONTAINER_PORT=9090 # To run 1.22.13 Kubernetes artifacts container on port 9090
 ```
 
-- `make build-container` is used to build the image builder container locally with all the dependencies like `Packer`, `Ansible`, and `OVF Tool`.
+- `make build-image-builder-container` is used to build the image builder container locally with all the dependencies like `Packer`, `Ansible`, and `OVF Tool`.
 
 ```bash
 make build-image-builder-container PRINT_HELP=y # To show the help information for this target.
-make build-image-builder-container              # To create the image builder container.
+make build-image-builder-container KUBERNETES_VERSION=v1.23.15+vmware.1 # To create the image builder container.
 ```
 
 - `make build-node-image` is used to build the vSphere Tanzu compatible node image for a Kubernetes version.

--- a/build-ova.sh
+++ b/build-ova.sh
@@ -15,14 +15,6 @@ artifacts_output_folder=${image_builder_root}/artifacts
 ova_destination_folder=${artifacts_output_folder}/ovas
 photon3_stig_compliance="false"
 
-function checkout_image_builder_branch() {
-    # Check out image builder with specific commit for the
-    # corresponding k8s version
-    cd ${image_builder_root}
-    git pull
-    git checkout ${IMAGE_BUILDER_COMMIT_ID}
-}
-
 function copy_custom_image_builder_files() {
     cp image/hack/tkgs-image-build-ova.py hack/image-build-ova.py
     cp image/hack/tkgs_ovf_template.xml hack/ovf_template.xml
@@ -123,7 +115,6 @@ function copy_ova() {
 }
 
 function main() {
-    checkout_image_builder_branch
     copy_custom_image_builder_files
     download_configuration_files
     generate_packager_configuration

--- a/docs/examples/adding_os_pkg_repos.md
+++ b/docs/examples/adding_os_pkg_repos.md
@@ -31,7 +31,7 @@ _**Note**: If you are building just the Ubuntu OS node image you will not have t
 
 [kubernetes image builder][kubernetes-image-builder] provides `extra_repos` packer variables through which sources/repositories can be configured for both Photon and Ubuntu. As there is a difference in how Ubuntu/Photon sources are configured we need to have separate source files for Photon/Ubuntu.
 
-- Create new folder `repos` in [ansible files](./../../../ansible/files/) folder
+- Create new folder `repos` in [ansible files][ansible-files] folder
 - Create a new file for Photon sources called `photon.repo` in the `repos` folder. Refer below for sample content and refer to the official Photon [document][photon-repo-doc] for more information
 
 ```text
@@ -100,9 +100,9 @@ To remove the extra repositories/sources that were configured during the image b
 
 [//]: Links
 
-[ansible-files]: [./../../../ansible/files/]
+[ansible-files]: ./../../ansible/files/
 [customizations-doc]: https://image-builder.sigs.k8s.io/capi/capi.html#customization
-[default-args]: [./../../../packer-variables/default-args.j2]
+[default-args]: ./../../packer-variables/default-args.j2
 [jinja]: https://jinja.palletsprojects.com/en/3.1.x/
 [kubernetes-image-builder]: https://github.com/kubernetes-sigs/image-builder/
 [photon-repo-doc]: https://vmware.github.io/photon/assets/files/html/3.0/photon_admin/adding-a-new-repository.html

--- a/hack/make-helpers/build-image-builder-container.sh
+++ b/hack/make-helpers/build-image-builder-container.sh
@@ -1,0 +1,29 @@
+#!/bin/bash
+# Copyright 2023 VMware, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+set -e
+
+source $(dirname "${BASH_SOURCE[0]}")/utils.sh
+enable_debugging
+
+is_argument_set "KUBERNETES_VERSION argument is required" $KUBERNETES_VERSION
+
+
+docker_build_args=$(jq -r '."'$KUBERNETES_VERSION'".docker_build_args | keys[]' $SUPPORTED_VERSIONS_JSON)
+build_variables=""
+for docker_arg in $docker_build_args;
+do
+    docker_arg_value=$(jq -r '."'$KUBERNETES_VERSION'".docker_build_args."'$docker_arg'"' $SUPPORTED_VERSIONS_JSON)
+    build_variables="${build_variables} --build-arg ${docker_arg}=${docker_arg_value}"
+done
+
+# by default don't show docker output
+docker_debug_flags="-q"
+if [ ! -z ${DEBUGGING+x} ]; then
+    docker_debug_flags="--progress plain"
+fi 
+
+docker build --platform=linux/amd64 $docker_debug_flags \
+-t $(get_image_builder_container_image_name $KUBERNETES_VERSION) \
+$build_variables $(dirname "${BASH_SOURCE[0]}")/../../.

--- a/hack/make-helpers/build-node-image.sh
+++ b/hack/make-helpers/build-node-image.sh
@@ -41,15 +41,12 @@ function build_node_image() {
 	-v $IMAGE_ARTIFACTS_PATH:/image-builder/images/capi/artifacts \
 	-w /image-builder/images/capi/ \
 	-e HOST_IP=$HOST_IP -e ARTIFACTS_CONTAINER_PORT=$ARTIFACTS_CONTAINER_PORT -e OS_TARGET=$OS_TARGET \
-	-e TKR_SUFFIX=$TKR_SUFFIX -e IMAGE_BUILDER_COMMIT_ID=$IMAGE_BUILDER_COMMIT_ID -e KUBERNETES_VERSION=$KUBERNETES_VERSION \
+	-e TKR_SUFFIX=$TKR_SUFFIX -e KUBERNETES_VERSION=$KUBERNETES_VERSION \
 	-e PACKER_HTTP_PORT=$PACKER_HTTP_PORT \
 	-p $PACKER_HTTP_PORT:$PACKER_HTTP_PORT \
 	--platform linux/amd64 \
-	$BYOI_IMAGE_NAME
+	$(get_image_builder_container_image_name $KUBERNETES_VERSION)
 }
-
-# Fetching Image Builder commit ID from supported-versions.json
-IMAGE_BUILDER_COMMIT_ID=$(jq -r '."'$KUBERNETES_VERSION'".extra_params.image_builder_commit' $SUPPORTED_VERSIONS_JSON)
 
 supported_os_list=$(jq -r '."'$KUBERNETES_VERSION'".supported_os' $SUPPORTED_VERSIONS_JSON)
 if [ "$supported_os_list" == "null" ]; then

--- a/hack/make-helpers/utils.sh
+++ b/hack/make-helpers/utils.sh
@@ -51,3 +51,14 @@ function get_node_image_builder_container_labels() {
     os_target=$2
     echo "-l byoi -l byoi_image_builder -l $kubernetes_version -l $os_target"
 }
+
+function get_image_builder_container_image_name() {
+    kubernetes_version=$1
+    default_image_name="vsphere-tanzu-byoi"
+    image_name=$default_image_name
+
+    # Docker image name doesn't support `+` so replace them with `---`
+    kubernetes_version=${kubernetes_version//+/---}
+    image_name=$image_name-$kubernetes_version
+    echo "$image_name"
+}

--- a/supported-versions.json
+++ b/supported-versions.json
@@ -1,23 +1,32 @@
 {
-    "v1.24.9+vmware.1" : {
-        "supported_os": ["photon-3", "ubuntu-2004-efi"],
+    "v1.24.9+vmware.1": {
+        "supported_os": [
+            "photon-3",
+            "ubuntu-2004-efi"
+        ],
         "artifacts_image": "projects.registry.vmware.com/tkg/tkg-vsphere-linux-resource-bundle:v1.24.9_vmware.1-tkg.1",
-        "extra_params": {
-            "image_builder_commit": "e5c6813621e13b3efc0f5181241673006fa643d9"
+        "docker_build_args": {
+            "IMAGE_BUILDER_COMMIT_ID": "e5c6813621e13b3efc0f5181241673006fa643d9"
         }
     },
-    "v1.25.7+vmware.3-fips.1" : {
-        "supported_os": ["photon-3", "ubuntu-2004-efi"],
+    "v1.25.7+vmware.3-fips.1": {
+        "supported_os": [
+            "photon-3",
+            "ubuntu-2004-efi"
+        ],
         "artifacts_image": "projects.registry.vmware.com/tkg/tkg-vsphere-linux-resource-bundle:v1.25.7_vmware.3-fips.1-tkg.1",
-        "extra_params": {
-            "image_builder_commit": "e5c6813621e13b3efc0f5181241673006fa643d9"
+        "docker_build_args": {
+            "IMAGE_BUILDER_COMMIT_ID": "e5c6813621e13b3efc0f5181241673006fa643d9"
         }
     },
-    "v1.26.5+vmware.2-fips.1" : {
-        "supported_os": ["photon-3", "ubuntu-2004-efi"],
+    "v1.26.5+vmware.2-fips.1": {
+        "supported_os": [
+            "photon-3",
+            "ubuntu-2004-efi"
+        ],
         "artifacts_image": "projects.registry.vmware.com/tkg/tkg-vsphere-linux-resource-bundle:v1.26.5_vmware.2-fips.1-tkg.1",
-        "extra_params": {
-            "image_builder_commit": "e5c6813621e13b3efc0f5181241673006fa643d9"
+        "docker_build_args": {
+            "IMAGE_BUILDER_COMMIT_ID": "e5c6813621e13b3efc0f5181241673006fa643d9"
         }
     }
 }

